src/shared-components/forms/index.js

```diff
@@ -0,0 +1,95 @@
+import PropTypes from 'prop-types';
+import React from 'react';
+import { TransitionGroup } from 'react-transition-group';
+
+import {
+  FieldContainer,
+  Label,
+  InputContainer,
+  Textarea,
+  Input,
+  ErrorItem,
+  HintItem,
+  HelperList,
+} from './style';
+import ErrorIcon from '../../svgs/icons/error-icon.svg';
+import HelperTransition from './helperTransition';
+
+class Field extends React.Component {
+  static propTypes = {
+    children: PropTypes.element.isRequired,
+    disabled: PropTypes.bool,
+    errors: PropTypes.objectOf(
+      PropTypes.oneOfType([
+        PropTypes.string,
+        PropTypes.arrayOf(PropTypes.string),
+      ])
+    ),
+    hintMessage: PropTypes.string,
+    label: PropTypes.string,
+    labelFor: PropTypes.string,
+  };
+
+  static defaultProps = {
+    disabled: false,
+    errors: {},
+    hintMessage: '',
+    label: '',
+  };
+
+  static Textarea = Textarea;
+
+  static Input = Input;
+
+  formatError = errorValue =>
+    Array.isArray(errorValue) ? errorValue.join(', ') : errorValue;
+
+  render() {
+    const {
+      children: inputChild,
+      disabled,
+      errors,
+      hintMessage,
+      label,
+      labelFor,
+    } = this.props;
+
+    const htmlFor = labelFor || label;
+    const errorKeys = Object.keys(errors);
+    const showErrors = errorKeys.length > 0;
+
+    return (
+      <FieldContainer>
+        {!!label && (
+          <Label htmlFor={htmlFor} disabled={disabled}>
+            {label}
+          </Label>
+        )}
+
+        <InputContainer showErrors={showErrors}>
+          <ErrorIcon />
+
+          {React.cloneElement(inputChild, {
+            disabled,
```

Comment:
Only concern here is if devs pass in more than one child. Not likely, but if they do it might break without them knowing why. If `cloneElement` throws an error stating you can only use one child, then I guess not a big deal.

Reply:
I actually researched the PropTypes docs for this, instaead of use `node` like everywhere else I am using `element` which force you to put only 1 child form the docs
https://reactjs.org/docs/typechecking-with-proptypes.html

```
Requiring Single Child
With PropTypes.element you can specify that only a single child can be passed to a component as children.
```

Reply:
👍